include/libtorrent/aux_/win_util.hpp

```diff
@@ -38,16 +38,40 @@ POSSIBILITY OF SUCH DAMAGE.
 
 namespace libtorrent { namespace aux {
 
+#ifdef TORRENT_WINRT
+	using LoadLibraryAPrototype = HMODULE WINAPI(_In_ LPCSTR lpLibFileName);
+#endif
+
 	template <typename Library>
 	HMODULE get_library_handle()
 	{
 		static bool handle_checked = false;
-		static HMODULE handle = nullptr;
+		static HMODULE handle;
 
 		if (!handle_checked)
 		{
-			handle = LoadLibraryA(Library::library_name);
 			handle_checked = true;
+
+#ifdef TORRENT_WINRT
+			MEMORY_BASIC_INFORMATION Information;
+
+			if (::VirtualQuery(&VirtualQuery, &Information, sizeof(Information)) == 0)
+			{
+				return nullptr;
+			}
+
+			const auto SyscallBegin = static_cast<HMODULE>(Information.AllocationBase);
+			const auto LoadLibraryA = reinterpret_cast<LoadLibraryAPrototype *>(::GetProcAddress(SyscallBegin, "LoadLibraryA"));
+
+			if (LoadLibraryA == nullptr)
+			{
+				return nullptr;
+			}
+
+			handle = LoadLibraryA(Library::library_name);
```

Comment:
did you observe this working in a WinRT build? Do `advapi32.dll`, `kernel32.dll` and `iphlpapi.dll` exist in that environment?

If not, and this will always fail, it might make sense to `#ifdef` this away higher up, where this utility is called

Reply:
Can confirm that the loads were successful, Desktop 19041 and Mobile 15254